### PR TITLE
New version 3.4.1, 3.5.0

### PR DESCRIPTION
**Pull request by maltesander**
# Description

- added support for versions 3.4.1, 3.5.0
- removed support for versions 2.7.1, 3.1.0, 3.2.0, 3.3.1

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [ ] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [x] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: CHANGELOG.md
Comment (siegfriedweber) [inline CHANGELOG.md:1]:
```diff
- - `vector` `0.26.0` -> `0.31.0` ([#612]).
+ - `vector` `0.26.0` -> `0.33.0` ([#612], [#627]).
```

**Comment by maltesander** [comment]
test: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/kafka-operator-it-custom/51/

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,21 +9,27 @@ All notable changes to this project will be documented in this file.
 - Default stackableVersion to operator version. It is recommended to remove `spec.image.stackableVersion` from your custom resources ([#611], [#613]).
 - Configuration overrides for the JVM security properties, such as DNS caching ([#616]).
 - Support PodDisruptionBudgets ([#625]).
+- Support new versions 3.4.1, 3.5.1 ([#627]).
```
Comment (adwk67) [inline CHANGELOG.md:12]:
```suggestion
- Support new versions 2.8.2, 3.4.1, 3.5.1 ([#627]).
```

**Review by adwk67 — CHANGES_REQUESTED** [review]
Just one small thing, othwerwise all good!

**Review by adwk67 — APPROVED** [review]
LGTM!

**Pull request merged**